watcher/tool/watcher/alerts.go

```diff
@@ -35,7 +35,7 @@ import (
 	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
 )
 
-func runAlertsWatcher(ctx context.Context, kubernetesClient *kubernetes.Client, retryC chan<- func() error) error {
+func runAlertsWatcher(config watcherConfig) error {
```

Comment:
My bad about suggesting to stash everything into config - it looks especially annoying with ctx hidden inside of it. Sorry - I think we can live with 4 parameters in the name of clarity!